javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+    val scalaRnd = scala.util.Random.javaRandomToRandom(rnd)
+    val shuffledList = scalaRnd.shuffle(_list.toSeq)
+    (0 until list.size).foreach(i => _list(i) = shuffledList(i))
+  }
+
+  def swap(list: List[_], i: Int, j: Int): Unit =
+    swapCore(list, i, j, true)
+
+  def fill[T](list: List[_ >: T], obj: T): Unit = {
+    @tailrec
+    def fillNext(iter: ListIterator[_ >: T]): Unit = {
+      if (iter.hasNext) {
+        iter.next()
+        iter.set(obj)
+        fillNext(iter)
+      }
+    }
+    fillNext(list.listIterator)
+  }
+
+  def copy[T](dest: List[_ >: T], src: List[_ <: T]): Unit = {
+    @tailrec
+    def copy(iterDest: ListIterator[_ >: T], iterSrc: ListIterator[_ <: T]): Unit = {
+      if (iterSrc.hasNext) {
+        if (iterDest.hasNext) {
+          iterDest.next()
+          iterDest.set(iterSrc.next())
+          copy(iterDest, iterSrc)
+        } else
+          throw new IndexOutOfBoundsException
+      }
+    }
+    copy(dest.listIterator, src.listIterator)
+  }
+
+  def min[T <: AnyRef with jl.Comparable[T]](coll: Collection[_ <: T]): T =
+    min(coll, makeComparator[T])
+
+  def min[T](coll: Collection[_ <: T], comp: Comparator[_ >: T]): T =
+    coll.min(comp)
+
+  def max[T <: AnyRef with jl.Comparable[T]](coll: Collection[_ <: T]): T =
+    max(coll, makeComparator[T])
+
+  def max[T](coll: Collection[_ <: T], comp: Comparator[_ >: T]): T =
+    coll.max(comp)
+
+  def rotate(list: List[_], distance: Int): Unit = {
+    val listSize = list.size
+    if (listSize > 1 && distance % listSize != 0) {
+      def exchangeRotation(): Unit = {
+        val _list = list.asInstanceOf[List[AnyRef]]
+        @tailrec
+        def rotateNext(index: Int, value: AnyRef): Unit = {
+          val nextValue = _list(index)
+          _list(index) = value
+          if (index != 0)
+            rotateNext(modulo(index + distance, listSize), nextValue)
+        }
+        rotateNext(modulo(distance, listSize), _list(0))
+      }
+
+      def splitReverseRotation(): Unit = {
+        val splitPoint = modulo(-distance, listSize)
+        reverse(list.subList(0, splitPoint))
+        reverse(list.subList(splitPoint, listSize))
+        reverse(list)
+      }
+
+      list match {
+        case _: RandomAccess    => exchangeRotation()
+        case _ if listSize < 16 => exchangeRotation() // TODO benchmark and set proper limit
+        case _                  => splitReverseRotation()
+      }
+    }
+  }
+
+  def replaceAll[T](list: List[T], oldVal: T, newVal: T): Boolean = {
+    @tailrec
+    def replaceAll(iter: ListIterator[T], mod: Boolean): Boolean = {
+      if (iter.hasNext) {
+        val isEqual = iter.next() === oldVal
+        if (isEqual)
+         iter.set(newVal)
+        replaceAll(iter, mod || isEqual)
+      } else {
+        mod
+      }
+    }
+    replaceAll(list.listIterator(), false)
+  }
+
+  def indexOfSubList(source: List[_], target: List[_]): Int =
+    indexOfSubListCore(source, target, true)
```

Comment:
Write `fromStart = true`. Typically we use named arguments when passing a brutal `true` or `false`.